src/main/java/com/wavefront/generators/TraceGenerator.java

```diff
@@ -96,8 +104,18 @@ protected void startGeneration(boolean isRealTime, @Nonnull GeneratorConfig gene
         }
       }
     }
+    logger.info("Generation complete!\n" + ANSI_YELLOW + String.format(generatorConfig.getGeneratorConfigFile() +
+            " Memory " +
+            "usage- %.2fGB / %.2fGB",
+        usedHeapMemoryGB, maxHeapMemoryGB) + ANSI_RESET);
     sendStat(str);
-    logger.info("Generation complete!");
+  }
+
+  private void updateHeapMemory() {
+    if (usedHeapMemoryGB < (double) memoryMXBean.getHeapMemoryUsage().getUsed() / CONVERT_BYTES_TO_GB) {
```

Comment:
Define 
double currentUsed = (double) memoryMXBean.getHeapMemoryUsage().getUsed() / CONVERT_BYTES_TO_GB;
and use this instead of two calculations. BTW, if calculates this two times you will have different values.